server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -301,24 +302,36 @@ public void removeResources(PurgeEvent event) {
 
   private void cleanupStorageSelectionCache(PurgeEvent event) {
     Function<String, Boolean> deleteConditionFunc = null;
+    String prefixKey = null;
     if (event instanceof AppPurgeEvent) {
+      prefixKey = UnionKey.buildKey(event.getAppId());
       deleteConditionFunc =
           partitionUnionKey -> UnionKey.startsWith(partitionUnionKey, event.getAppId());
     } else if (event instanceof ShufflePurgeEvent) {
+      assert event.getShuffleIds().size() == 1; // only one shuffleId
+      int shuffleId = event.getShuffleIds().get(0);
+      prefixKey = UnionKey.buildKey(event.getAppId(), shuffleId);
       deleteConditionFunc =
-          partitionUnionKey ->
-              UnionKey.startsWith(partitionUnionKey, event.getAppId(), event.getShuffleIds());
+          partitionUnionKey -> UnionKey.startsWith(partitionUnionKey, event.getAppId(), shuffleId);
     }
     long startTime = System.currentTimeMillis();
-    deleteElement(partitionsOfStorage, deleteConditionFunc);
+    deleteElement(partitionsOfStorage.tailMap(prefixKey), deleteConditionFunc);
     LOG.info(
         "Cleaning the storage selection cache costs: {}(ms) for event: {}",
         System.currentTimeMillis() - startTime,
         event);
   }
 
   private <K, V> void deleteElement(Map<K, V> map, Function<K, Boolean> deleteConditionFunc) {
```

Comment:
Could you help rename the `map` to `sortedPartitionsOfStorageMap`? If you don't mind, the comments could be added in this method. Because the deleting logic depends on the sorted map.